osu.Game/Users/UserRankPanel.cs

```diff
@@ -0,0 +1,212 @@
+{
+    /// <summary>
+    /// User card that shows user's global and country ranks in the bottom.
+    /// Meant to be used in the toolbar login overlay.
+    /// </summary>
+    public partial class UserRankPanel : UserPanel
+    {
+        private const int padding = 10;
+        private const int main_content_height = 80;
+
+        public UserRankPanel(APIUser user)
+            : base(user)
+        {
+            AutoSizeAxes = Axes.Y;
+            CornerRadius = 10;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            BorderColour = ColourProvider?.Light1 ?? Colours.GreyVioletLighter;
+        }
+
+        protected override Drawable CreateLayout()
+        {
+            FillFlowContainer details;
+
+            var layout = new Container
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Children = new Drawable[]
+                {
+                    new Container
+                    {
+                        Name = "Main content",
+                        RelativeSizeAxes = Axes.X,
+                        Height = main_content_height,
+                        CornerRadius = 10,
+                        Masking = true,
+                        Children = new Drawable[]
+                        {
+                            new UserCoverBackground
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                User = User,
+                                Alpha = 0.3f
+                            },
+                            new GridContainer
+                            {
+                                AutoSizeAxes = Axes.Y,
+                                RelativeSizeAxes = Axes.X,
+                                ColumnDimensions = new[]
+                                {
+                                    new Dimension(GridSizeMode.Absolute, padding),
+                                    new Dimension(GridSizeMode.AutoSize),
+                                    new Dimension(),
+                                    new Dimension(GridSizeMode.Absolute, padding),
+                                },
+                                RowDimensions = new[]
+                                {
+                                    new Dimension(GridSizeMode.Absolute, padding),
+                                    new Dimension(GridSizeMode.AutoSize),
+                                },
+                                Content = new[]
+                                {
+                                    new[]
+                                    {
+                                        Empty(),
+                                        Empty(),
+                                        Empty(),
+                                        Empty()
```

Comment:
what on earth.

Reply:
Padding, similar to how `UserGridPanel` does it

Reply:
Please add an inline comment explaining this.